src/tests/ftest/util/apricot/apricot/test.py

```diff
@@ -78,24 +83,32 @@ def __init__(self, *args, **kwargs):
         """Initialize a Test object."""
         super(Test, self).__init__(*args, **kwargs)
 
-        # if self.timeout is a string any time value may be entered
-        # but the time speciified must be in this order DHMS
-        # examples   10D2H10M10S or 25M or 2H5S
-        # pylint: disable=no-member
+        # Support specifying timeout values with units, e.g. "1d 2h 3m 4s".
+        # Any unit combination may be used, but they must be specified in
+        # descending order. Spaces can optionally be used between units and
+        # values. The first unit character is required; other unit characters
+        # are optional. The units are case-insensitive.
+        # The following units are supported:
+        #   - days      e.g. 1d, 1 day
+        #   - hours     e.g. 2h, 2 hrs, 2 hours
+        #   - minutes   e.g. 3m, 3 mins, 3 minutes
+        #   - seconds   e.g. 4s, 4 secs, 4 seconds
         if isinstance(self.timeout, str):
             pattern = r""
             for interval in ("days", "hours", "minutes", "seconds"):
                 pattern += r"(?:(\d+)(?:\s*{0}[{1}]*\s*)){{0,1}}".format(
                     interval[0], interval[1:])
+            # pylint: disable=no-member
             dhms = re.search(pattern, self.timeout, re.IGNORECASE).groups()
+            # pylint: enable=no-member
             self.timeout = 0
             for index, multiplier in enumerate([24 * 60 * 60, 60 * 60, 60, 1]):
                 if dhms[index] is not None:
                     self.timeout += multiplier * int(dhms[index])
```

Comment:
Why not place this it's own function? or even place it in general_utils?

Reply:
It could be a function in general_utils.py.   Currently I believe this is the only place it is used.